### PR TITLE
DP-1232 One handler event on list.set()

### PR DESCRIPTION
**Pull request by aleksei-berezkin**
This is a member of a triple pull request. This one may look independent, but I didn't tested it separately from others, so please merge all of them or neither.

Dependent pull requests to be merged synchronously:
https://github.com/JetBrains/jetpad-projectional/pull/275
https://github.com/JetBrains/datapad/pull/36

### CODE REVIEW

**Comment by glebleonov** [comment]
I see 11 usages of CollectionItemEvent.SET, but 40 usages of ADD and 32 usages of REMOVE. All these usages should be inspected and, most likely, refactored. @aleksei-berezkin, did you perform such work?

File: model/src/main/java/jetbrains/jetpad/model/collections/CollectionEventHandlers.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package jetbrains.jetpad.model.collections;
+
+import jetbrains.jetpad.model.event.EventHandler;
+
+public class CollectionEventHandlers {
```
Comment (glebleonov) [inline model/src/main/java/jetbrains/jetpad/model/collections/CollectionEventHandlers.java]: I don't see any objection to place this class in test sources.

Comment (aleksei-berezkin) [inline model/src/main/java/jetbrains/jetpad/model/collections/CollectionEventHandlers.java]: Yes, I use them in other modules tests.

**Comment by aleksei-berezkin** [comment]
I checked all usages of `CollectionItemEvent`, and the only place requiring changes was in Census (plz see dependent datapad request). Other places either regarded all three events fairly, or used collection adapters which is safe.

File: model/src/test/java/jetbrains/jetpad/model/collections/ObservableArrayListTest.java [hunk 2]
```diff
@@ -118,6 +119,15 @@ public void iteratorRemove() {
     assertEvent(0, "xyz", null, CollectionItemEvent.EventType.REMOVE);
   }
 
+  @Test
+  public void oneHandlerEventOnSet() {
+    addSampleItem();
+    CollectionEventHandlers.CountingHandler<String> counter = new CollectionEventHandlers.CountingHandler<>();
```
Comment (glebleonov) [inline model/src/test/java/jetbrains/jetpad/model/collections/ObservableArrayListTest.java]: It is better to check for one set event explicitly.

**Comment by aleksei-berezkin** [comment]
Checked usages of ADD, REMOVE and SET, found two places requiring changes, both of them in DataPad, updated the corresponding request.

Changes in last commit of this request: moved counting handler to test sources, added event kind check.

**Comment by glebleonov** [comment]
Ok, prepare a squashed pull request, please.

**Comment by aleksei-berezkin** [comment]
Please see one-commit request #84.

**Pull request closed without merging**